x/axelarnet/types/evm_wasm_translator.go

```diff
@@ -0,0 +1,90 @@
+package types
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+
+	"github.com/axelarnetwork/utils/funcs"
+)
+
+var (
+	stringType      = funcs.Must(abi.NewType("string", "string", nil))
+	stringArrayType = funcs.Must(abi.NewType("string[]", "string[]", nil))
+	bytesType       = funcs.Must(abi.NewType("bytes", "bytes", nil))
+
+	// abi encoded bytes, with the following format:
+	// wasm method name, argument type list, encoded bytes contain actual argument names and values
+	payloadArguments = abi.Arguments{{Type: stringType}, {Type: stringArrayType}, {Type: bytesType}}
+)
+
+type contract struct {
+	// Contract is the address of the wasm contract
+	Contract string `json:"contract"`
+	// Msg is a json struct {"methodName": {"arg1": "val1", "arg2": "val2"}}
+	Msg map[string]interface{} `json:"msg"`
+}
+
+// wasm is the json get passed to the IBC memo field
+type wasm struct {
+	Wasm contract `json:"wasm"`
+}
+
+// ConstructWasmMessage creates a json serialized wasm message from Axelar defined abi encoded payload
```

Comment:
can you expand this a bit? I find the description a bit confusing and it's probably no use to anybody who doesn't already know how the process works